tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/rules/FunCtorRule.scala

```diff
@@ -101,28 +99,20 @@ class FunCtorRule(rewriter: SymbStateRewriter) extends RewritingRule {
       state: SymbState,
       mapEx: TlaEx,
       varName: String,
-      setEx: TlaEx,
-      oldCells: Seq[ArenaCell]) = {
-    // similar to SymbStateRewriter.rewriteSeqUntilDone and SetFilterRule
-    def process(st: SymbState, seq: Seq[ArenaCell]): (SymbState, Seq[TlaEx]) = {
-      seq match {
-        case Seq() =>
-          (st, List())
+      oldCells: Seq[ArenaCell]): (SymbState, Seq[ArenaCell]) = {
+    var nextState = state
 
-        case cell :: tail =>
-          val (ts: SymbState, nt: List[TlaEx]) = process(st, tail)
-          val newBinding = Binding(ts.binding.toMap + (varName -> cell))
-          val cellState = new SymbState(mapEx, ts.arena, newBinding)
-          // add [cell/x]
-          val ns = rewriter.rewriteUntilDone(cellState)
-          (ns.setBinding(ts.binding), ns.ex :: nt) // reset binding and add the new expression in the head
-      }
+    // similar to SymbStateRewriter.rewriteSeqUntilDone and SetFilterRule
```

Comment:
This comment could be improved. I don't know `rewriteSeqUntilDone` and `SetFilterRule`. Even if I did, what does "similar" mean in this context?

Reply:
Good point. This comment seems to be outdated. Removed.